examples/Example/src/new_api/calculator/index.tsx

```diff
@@ -0,0 +1,399 @@
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+  runOnJS,
+} from 'react-native-reanimated';
+
+const DRAG_ANIMATION_DURATION = 300;
+const TAP_ANIMATION_DURATION = 100;
+const window = Dimensions.get('window');
+
+export default function CalculatorUI() {
+  const outputOffset = useSharedValue(0);
+  const [history, setHistory] = useState(Array<string>());
+  const [expression, setExpression] = useState('');
+
+  function measure(e: LayoutChangeEvent) {
+    outputOffset.value = -e.nativeEvent.layout.height;
+  }
+
+  return (
+    <View style={styles.home}>
+      <Output offset={outputOffset} expression={expression} history={history} />
+      <Input
+        offset={outputOffset}
+        measure={measure}
+        expression={expression}
+        setExpression={setExpression}
+        setHistory={setHistory}
+      />
+    </View>
+  );
+}
+
+function Output(props: {
+  offset: Animated.SharedValue<number>;
+  expression: string;
+  history: string[];
+}) {
+  const layout = useRef({});
+  const scrollView = useRef<ScrollView>();
+  const drag = useSharedValue(0);
+  const dragOffset = useSharedValue(0);
+  const [opened, setOpened] = useState(false);
+
+  function measure(e: LayoutChangeEvent) {
+    layout.current = e.nativeEvent.layout;
+  }
+
+  function open() {
+    drag.value = withTiming(-props.offset.value, {
+      duration: DRAG_ANIMATION_DURATION,
+    });
+    dragOffset.value = -props.offset.value;
+
+    setOpened(true);
+  }
+
+  function close() {
+    drag.value = withTiming(0, { duration: DRAG_ANIMATION_DURATION });
+    dragOffset.value = 0;
+
+    setOpened(false);
+  }
+
+  const translationStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateY: props.offset.value + drag.value }],
+    };
+  });
+
+  const dragGesture = useAnimatedGesture(
+    Gesture.pan()
+      .setOnUpdate((e) => {
+        'worklet';
+        const value = dragOffset.value + e.translationY;
+
+        if (value > -props.offset.value) {
+          drag.value = -props.offset.value;
+        } else if (value < 0) {
+          drag.value = 0;
+        } else {
+          drag.value = value;
+        }
+      })
+      .setOnEnd((e) => {
+        'worklet';
+        const value = dragOffset.value + e.translationY;
+
+        if (opened) {
+          if (value < -props.offset.value - 100) {
+            runOnJS(close)();
+          } else {
+            runOnJS(open)();
+          }
+        } else {
+          if (value > 100) {
+            runOnJS(open)();
+          } else {
+            runOnJS(close)();
+          }
+        }
+      })
+  );
+
+  scrollView.current?.scrollToEnd({ animated: true });
+
+  return (
+    <GestureMonitor gesture={dragGesture}>
+      <Animated.View
+        style={[styles.output, translationStyle]}
+        onLayout={measure}>
+        <ScrollView
+          ref={(ref: ScrollView) => {
+            if (!opened && ref) ref.scrollToEnd({ animated: false });
```

Comment:
```suggestion
            if (!opened && ref) { ref.scrollToEnd({ animated: false }); }
```

Reply:
Fixed in 3a5fca0b55a2d504c90c4cb9449f48ce4ebefced